mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -528,10 +538,22 @@ object ALS extends Logging {
         itemFactors = computeFactors(userFactors, userOutBlocks, itemInBlocks, rank, regParam,
           userLocalIndexEncoder, implicitPrefs, alpha, solver)
         previousItemFactors.unpersist()
-        if (sc.checkpointDir.isDefined && (iter % 3 == 0)) {
+        itemFactors.setName(s"itemFactors-$iter").persist(intermediateRDDStorageLevel)
+        // TODO: Generalize PeriodicGraphCheckpointer and use it here.
+        if (sc.checkpointDir.isDefined && (iter % checkpointInterval == 0)) {
           itemFactors.checkpoint()
+          itemFactors.count() // checkpoint item factors and cut lineage
```

Comment:
Ah, for implicit preference, this is not necessary because we are computing YtY anyway.